docs/Makefile

```diff
@@ -0,0 +1,200 @@
+# Results are sorted and without trailing path separator
+# $1 -> Root paths
+define ListSubDirs
+$(sort $(foreach d,$(dir $(wildcard $1/*/.)),$(d:/=)))
+endef
+
+SRC_INCDIR			:= $(SOURCEDIR)/_inc
+
+# $1 -> directory list
+define FindReadme
+$(foreach d,$1,$(wildcard $d/READ*.md $d/Read*.md $d/read*.md $d/read*.txt $d/*.rst))
+endef
+
+# $1 -> directory list
+define FindImages
+$(foreach d,$1,$(wildcard $d/*.svg $d/*.png $d/*.jpg))
+endef
+
+# Each sample MUST have a README.rst file
+SAMPLE_DIRS		:= $(call ListSubDirs,$(SMINGDIR)/samples)
+SOURCE_FILES 	+= \
+	$(addsuffix /README.rst,$(SAMPLE_DIRS)) \
+	$(call FindImages,$(SAMPLE_DIRS))
+
+LIBRARY_DIRS	:= $(call ListSubDirs,$(SMING_HOME)/Libraries)
+LIBRARY_DOCS 	:= $(call FindReadme,$(LIBRARY_DIRS))
+LIBRARY_INDEX	:= $(SRC_INCDIR)/Sming/Libraries/index.rst
+SOURCE_FILES	+= \
+	$(LIBRARY_DOCS) \
+	$(call FindImages,$(LIBRARY_DIRS)) \
+	Sming/Libraries/ArduinoJson5/ArduinoJson/README.md \
+	Sming/Libraries/ArduinoJson5/ArduinoJson/banner.svg \
+   	Sming/Libraries/ArduinoJson6/ArduinoJson/README.md \
+	Sming/Libraries/ArduinoJson6/ArduinoJson/banner.svg
+
+COMPONENT_DIRS		:=
+# $1 -> Arch name
+define FindComponentDocs
+COMPONENT_SRC_$1	:= $(SMING_HOME)/$(if $1,Arch/$1/)Components
+COMPONENT_DIRS_$1	= $$(call ListSubDirs,$$(COMPONENT_SRC_$1))
+COMPONENT_DIRS		+= $$(COMPONENT_DIRS_$1)
+COMPONENT_DOCS_$1	= $$(call FindReadme,$$(COMPONENT_DIRS_$1))
+COMPONENT_DOCS 		+= $(patsubst $$(COMPONENT_SRC_$1)/%,Components/$(if $1,$1/)%,$$(COMPONENT_DOCS_$1))
+ifeq ($V,1)
+$$(info COMPONENT_SRC_$1 = $$(COMPONENT_SRC_$1))
+$$(info COMPONENT_DIRS_$1 = $$(COMPONENT_DIRS_$1))
+$$(info COMPONENT_DOCS_$1 = $$(COMPONENT_DOCS_$1))
+endif
+endef
+
+COMPONENT_DOCS		:= $(SMING_HOME)/README.rst
+$(eval $(call FindComponentDocs,))
+$(eval $(call FindComponentDocs,Esp8266))
+$(eval $(call FindComponentDocs,Host))
+
+ifeq ($V,1)
+$(info COMPONENT_DOCS = $(COMPONENT_DOCS))
+endif
+
+COMPONENT_INDEX	:= $(SRC_INCDIR)/Components/index.rst
+SOURCE_FILES	+= $(COMPONENT_DOCS)
+
+# This macro sets the default component variables before including the (optional) component.mk file.
+# $1 -> Component name
+# $2 -> Component path
+define ParseComponent
+$(if $V,$(info -- Parsing $2))
+# Process any component.mk file (optional)
+ifneq (,$(wildcard $2/component.mk))
+COMPONENT_RULE			:= __no_build__
+COMPONENT_NAME			:= $1
+COMPONENT_PATH			:= $2
+COMPONENT_VARS			:=
+CONFIG_VARS				:=
+RELINK_VARS				:=
+DEBUG_VARS				:=
+COMPONENT_DOCFILES		:=
+COMPONENT_DEPENDS		:=
+ENABLE_CUSTOM_LWIP		:= 2
+include $2/component.mk
+CMP_$1_DEPENDS			:= $$(COMPONENT_DEPENDS)
+CMP_$1_VARS				:= $$(sort $$(COMPONENT_VARS))
+SOURCE_FILES			+= $$(addprefix $$(COMPONENT_PATH)/,$$(COMPONENT_DOCFILES))
+endif
+endef # ParseComponent
+
+$(foreach d,$(COMPONENT_DIRS),$(eval $(call ParseComponent,$(notdir $d),$d)))
+
+
+#
+SOURCE_FILES	:= $(patsubst $(SMINGDIR)/%,%,$(SOURCE_FILES))
+
+SOURCE_DIRS		:= $(patsubst %/,$(SRC_INCDIR)/%,$(sort $(dir $(SOURCE_FILES))))
```

Comment:
With the current code I ended up with the following error:
`make[1]: *** No rule to make target 'source/_inc/', needed by 'source/_inc/CONTRIBUTING.md'.  Stop.`
changing the line above to:
`SOURCE_DIRS		:= $(addsuffix /, $(patsubst %/,$(SRC_INCDIR)/%,$(sort $(dir $(SOURCE_FILES)))))`
fixed the issue(s) for me.